functions/New-DbaFirewallRule.ps1

```diff
@@ -0,0 +1,281 @@
+                try {
+                    $commandResult = Invoke-Command2 -ComputerName $instance.ComputerName -Credential $Credential -ScriptBlock $cmdScriptBlock -ArgumentList $config
+                } catch {
+                    Stop-Function -Message "Failed to execute command on $($instance.ComputerName) for instance $($instance.InstanceName)." -Target $instance -ErrorRecord $_ -Continue
+                }
+
+                # Output information
+                [PSCustomObject]@{
+                    ComputerName = $instance.ComputerName
+                    InstanceName = $instance.InstanceName
+                    SqlInstance  = $instance.SqlFullName.Trim('[]')
+                    DisplayName  = $config['DisplayName']
+                    Successful   = $commandResult.Successful
+                    Status       = $commandResult.CimInstance.Status
+                    Warning      = $commandResult.Warning
+                    Error        = $commandResult.Error
+                    Exception    = $commandResult.Exception
+                    Details      = $commandResult
+                } | Select-DefaultView -Property ComputerName, InstanceName, SqlInstance, DisplayName, Successful, Status, Warning, Error, Exception
+            }
+
+            # Firewall rule for the instance is in place, let's see if we need one for the SQL Server Browser
+            if ($browserNeeded) {
+                # Apply the defaults
+                $config = @{
+                    DisplayName = 'SQL Server Browser'
+                    Group       = 'SQL Server'
+                    Enabled     = 'True'
+                    Direction   = 'Inbound'
+                    Protocol    = 'UDP'
+                    LocalPort   = 1434
+                }
+
+                # Apply the given configuration
+                if ($Configuration) {
+                    foreach ($param in $Configuration.Keys) {
+                        $config[$param] = $Configuration[$param]
+                    }
+                }
+
+                # Run the command for the browser
+                if ($PSCmdlet.ShouldProcess($instance, "Creating firewall rule for SQL Server Browser on $($instance.ComputerName)")) {
+                    try {
+                        $commandResult = Invoke-Command2 -ComputerName $instance.ComputerName -Credential $Credential -ScriptBlock $cmdScriptBlock -ArgumentList $config
+                    } catch {
+                        Stop-Function -Message "Failed to execute command on $($instance.ComputerName) for instance $($instance.InstanceName)." -Target $instance -ErrorRecord $_ -Continue
+                    }
+
+                    # Output information
+                    [PSCustomObject]@{
+                        ComputerName = $instance.ComputerName
+                        InstanceName = $instance.InstanceName
+                        SqlInstance  = $instance.SqlFullName.Trim('[]')
```

Comment:
omg does this remove the start `[` and `]`?! That's awesome! I always do replaces.

Reply:
Yes, it does. It would be very nive if our [DbaInstanceParameter] type would have a property "SqlInstance" for those use cases...